storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/RocksDbConfiguration.java

```diff
@@ -13,46 +13,86 @@
+ * declarations is compatible with existing databases. These values will be used if the field didn't
+ * exist at the time the database was created, so typically should match the RocksDB default.
+ *
+ * <p>If the value to use for new databases, differs from the original, set it in a factory function
+ * e.g. {@link #v5ArchiveDefaults()}.
+ *
+ * <p>Values that are safe to change for existing databases are marked with {@link
+ * Access#WRITE_ONLY}. They will not be written to the metadata file but if present, the value will
+ * be loaded providing a simple way to experiment with different values without it being fixed at
+ * database creation.
+ */
+@SuppressWarnings("FieldMayBeFinal")
+@JsonIgnoreProperties(ignoreUnknown = true)
 public class RocksDbConfiguration {
   public static final int DEFAULT_MAX_OPEN_FILES = 1024;
   public static final int DEFAULT_MAX_BACKGROUND_COMPACTIONS = 4;
   public static final int DEFAULT_BACKGROUND_THREAD_COUNT = 4;
   public static final long DEFAULT_CACHE_CAPACITY = 8 << 20;
   public static final long DEFAULT_WRITE_BUFFER_CAPACITY = 128 << 20;
 
-  private final Path databaseDir;
-  private final int maxOpenFiles;
-  private final int maxBackgroundCompactions;
-  private final int backgroundThreadCount;
-  private final long cacheCapacity;
-  private final long writeBufferCapacity;
-
-  public RocksDbConfiguration(
-      Path databaseDir,
-      int maxOpenFiles,
-      int maxBackgroundCompactions,
-      int backgroundThreadCount,
-      long cacheCapacity,
-      long writeBufferCapacity) {
-    this.databaseDir = databaseDir;
-    this.maxOpenFiles = maxOpenFiles;
-    this.maxBackgroundCompactions = maxBackgroundCompactions;
-    this.backgroundThreadCount = backgroundThreadCount;
-    this.cacheCapacity = cacheCapacity;
-    this.writeBufferCapacity = writeBufferCapacity;
+  /* --------------- Safe to Change Properties ------------ */
+
+  @JsonProperty(value = "maxOpenFiles", access = Access.WRITE_ONLY)
+  private int maxOpenFiles = DEFAULT_MAX_OPEN_FILES;
+
+  @JsonProperty(value = "maxBackgroundCompactions", access = Access.WRITE_ONLY)
+  private int maxBackgroundCompactions = DEFAULT_MAX_BACKGROUND_COMPACTIONS;
+
+  @JsonProperty(value = "backgroundThreadCount", access = Access.WRITE_ONLY)
+  private int backgroundThreadCount = DEFAULT_BACKGROUND_THREAD_COUNT;
+
+  @JsonProperty(value = "cacheCapacity", access = Access.WRITE_ONLY)
+  private long cacheCapacity = DEFAULT_CACHE_CAPACITY;
+
+  @JsonProperty(value = "writeBufferCapacity", access = Access.WRITE_ONLY)
+  private long writeBufferCapacity = DEFAULT_WRITE_BUFFER_CAPACITY;
+
+  /* ---------------     Fixed Properties     ------------ */
+
+  @JsonProperty("compressionType")
+  private CompressionType compressionType = CompressionType.NO_COMPRESSION;
+
+  @JsonProperty("bottomMostCompressionType")
+  private CompressionType bottomMostCompressionType = CompressionType.NO_COMPRESSION;
```

Comment:
not the hugest fan of bottomMost... lowest / minimum?

Reply:
It's a RocksDB term - the DB is conceptually made up of a series of layers stacked on top of each other with data gradually being pushed down to the bottom layer as it's been unchanged for longer periods.  Hence, bottom-most.